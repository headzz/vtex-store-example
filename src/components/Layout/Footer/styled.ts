import { colors } from 'configuration'
import styled from 'styled-components'

export const FooterStyled = styled.section`
  background-color: ${colors.BLACK};
  padding: 2.4rem 0rem;
  width: 100%;

  @media (min-width: 768px) {
    padding: 2.4rem 20.8rem;
  }
`
