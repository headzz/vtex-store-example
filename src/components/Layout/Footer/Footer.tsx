import { FooterStyled } from './styled'

/**
 * @component
 * @name Footer
 * @description Defines the container for the layout of footer section.
 * @namespace components components/Layout/Footer
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Footer: React.FC = ({ children }) => {
  return <FooterStyled>{children}</FooterStyled>
}

export default Footer
