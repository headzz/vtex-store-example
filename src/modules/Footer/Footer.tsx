import Flexbox from 'components/Flexbox/Flexbox'
import Layout from 'components/Layout'
import Link from 'components/Link'
import Image from 'next/image'
import { FooterContainer } from './styled'

/**
 * @component
 * @name Footer
 * @description creates Footer component section to use on screen.
 * @namespace components modules/Footer
 *
 */

const Footer: React.FC = () => (
  <Layout.Footer>
    <FooterContainer>
      <Link background={false} margin={false} href="/" border={false}>
        <Image
          src="/img/wppcompany.png"
          height={27}
          width={142}
          alt={'logo do grupo wpp'}
        />
      </Link>

      <Flexbox alignContent="center">
        <Link background={false} margin={false} href="/" border={false}>
          <Image
            src="/img/social/facebook.png"
            height={32}
            width={32}
            alt="rede social facebook"
            layout="fixed"
          />
        </Link>
        <Link background={false} margin={false} href="/" border={false}>
          <Image
            src="/img/social/instagram.png"
            height={32}
            width={32}
            alt="rede social instagram"
            layout="fixed"
          />
        </Link>
        <Link background={false} margin={false} href="/" border={false}>
          <Image
            src="/img/social/linkedin.png"
            height={32}
            width={32}
            alt="rede social linkedin"
            layout="fixed"
          />
        </Link>
      </Flexbox>
    </FooterContainer>
  </Layout.Footer>
)

export default Footer
