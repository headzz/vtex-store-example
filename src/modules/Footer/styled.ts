import styled from 'styled-components'

export const FooterContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 100%;

  div:nth-child(2) {
    width: 100%;
  }
  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;

    div:nth-child(2) {
      width: 13rem;
    }
  }
`
