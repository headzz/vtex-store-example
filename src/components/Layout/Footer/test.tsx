import { render } from '@testing-library/react'

import Footer from './Footer'

describe('<Footer />', () => {
  it('should render component that wraps Footer on page', () => {
    const { container } = render(<Footer />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
