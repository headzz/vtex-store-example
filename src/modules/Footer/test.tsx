import { render } from '@testing-library/react'

import Footer from './Footer'

describe('Footer', () => {
  it('match snapshot', () => {
    const { container } = render(<Footer />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
