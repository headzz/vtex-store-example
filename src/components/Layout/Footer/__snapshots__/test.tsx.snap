// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render component that wraps Footer on page 1`] = `
.c0 {
  background-color: #000000;
  padding: 2.4rem 0rem;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem 20.8rem;
  }
}

<section
  class="c0"
/>
`
